addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/internal/core/LxControlColorPickerV2.java

```diff
@@ -0,0 +1,294 @@
+
+    LxControlColorPickerV2(LxWsClient client, LxUuid uuid, LxJsonControl json, LxContainer room, LxCategory category) {
+        super(client, uuid, json, room, category);
+    }
+
+    /**
+     * get the current Loxone color in HSBType format
+     *
+     * @return the HSBType color
+     */
+    public HSBType getColor() {
+        HSBType hsbColor = null;
+        String color = getStateTextValue(STATE_COLOR);
+
+        if (color != null) {
+            hsbColor = this.mapLoxoneToOH(color);
+        }
+
+        return hsbColor;
+    }
+
+    /**
+     * sets the color of the color picker
+     *
+     * @param hsb the color to set
+     */
+    public void setColor(HSBType hsb) {
+        socketClient.sendAction(uuid, "hsv(" + hsb.toString() + ")");
+    }
+
+    /**
+     * Sets the color picker to on
+     *
+     * @throws IOException
+     *                         error communicating with the Miniserver
+     */
+    public void on() throws IOException {
+        HSBType currentColor = this.getColor();
+        if (currentColor != null) {
+            this.setColor(new HSBType(currentColor.getHue(), currentColor.getSaturation(), PercentType.HUNDRED));
+        }
+    }
+
+    /**
+     * Sets the color picker to off
+     *
+     * @throws IOException
+     *                         error communicating with the Miniserver
+     */
+    public void off() throws IOException {
+        HSBType currentColor = this.getColor();
+        if (currentColor != null) {
+            this.setColor(new HSBType(currentColor.getHue(), currentColor.getSaturation(), PercentType.ZERO));
+        }
+    }
+
+    /**
+     * set the brightness level
+     *
+     * @param p the brightness percentage
+     */
+    public void setBrightness(PercentType p) {
+        HSBType currentColor = this.getColor();
+        if (currentColor != null) {
+            this.setColor(new HSBType(currentColor.getHue(), currentColor.getSaturation(), p));
+        }
+    }
+
+    /**
+     * set the brightness level from a decimal type
+     *
+     * @param d the brightness in decimal
+     */
+    public void setBrightness(DecimalType d) {
+        this.setBrightness(new PercentType(d.toBigDecimal()));
+    }
+
+    /**
+     * increases/decreases the brightness with a given step
+     *
+     * @param step the amount to increase/decrease
+     */
+    public void increaseDecreaseBrightness(int step) {
+        HSBType currentColor = this.getColor();
+        if (currentColor != null) {
+            this.setBrightness(new PercentType(currentColor.getBrightness().intValue() + step));
+        }
+    }
+
+    /**
+     * map the Loxone color to OpenHab HSBType
+     *
+     * @param color color in format hsb(h,s,v) or temp(brightness,kelvin)
+     * @return HSBType
+     */
+    private HSBType mapLoxoneToOH(String color) {
+        HSBType hsbColor = null;
+
+        Matcher m = Pattern.compile("\\((.*?)\\)").matcher(color);
+        if (m.find()) {
```

Comment:
Ok, this way of matching the syntax will effectively work, but it looks a bit like "hacking". Specifically for example hsvabc(1,2,3) will get accepted. Also hsvabc(a,b,c) or hsv(,,,) will get accepted too, where the values are not numbers and it will get passed to the Integer.valueOf causing exception. I think for the sake of purity it would be good to match the syntax of the received command against the exact format that is expected, separately to hsv and to temp. For example hsv((\d+),(\d+),(\d+)) and take each group member (this is not exact regular expression that would work, just an idea example).